src/runtime/cpu_device_api.cc

```diff
@@ -32,8 +35,20 @@ class CPUDeviceAPI final : public DeviceAPI {
     ptr = memalign(alignment, nbytes);
     if (ptr == nullptr) throw std::bad_alloc();
 #else
-    int ret = posix_memalign(&ptr, alignment, nbytes);
-    if (ret != 0) throw std::bad_alloc();
+#ifndef _WIN32
+    constexpr size_t _HugePage2MB_ = 1<<21;
+    if (nbytes >= _HugePage2MB_) {
```

Comment:
This condition seems be insufficient. If nbytes is only a bit more than _HugePage2MB_, e.g. _HugePage2MB_+64, we are likely losing space by allocating huge table for small number of bytes and, in consequence, the time because all allocated pages have to be cleared in KMD for security reason.
What about making decision based on proportion of wasted memory to real needs?
For example:
size_t wasted = _HugePage2MB_ - (nbytes % _HugePage2MB_);
float procwasted = (float)wasted/nbytes;
if (procwated < 0.2) {
....